src/Sentry/Internal/Http/RateLimitCategory.cs

```diff
@@ -17,13 +17,18 @@ public bool Matches(EnvelopeItem item)
             return true;
         }
 
-        var type = item.TryGetType();
+        var type = item.TryGetType()?.ToLower();
         if (string.IsNullOrWhiteSpace(type))
         {
             return false;
         }
 
-        return string.Equals(Name, type, StringComparison.OrdinalIgnoreCase);
+        // The envelope item type for metrics is `statsd`, but the client report category is `metric_bucket`
+        var reportCategory = EnvelopeItem.TypeValueMetric == type
+            ? "metric_bucket"
+            : type;
+
+        return string.Equals(Name, reportCategory, StringComparison.OrdinalIgnoreCase);
```

Comment:
Can we be a bit more opinionated here? We're now allocating a new string, comparing multiple times in different ways just to special case the `metric_bucket -> statsd` conversion.
And that happens for every category.

Reply:
I've replaced with a switch that removes the string allocation.